arviz/plots/backends/matplotlib/ecdfplot.py

```diff
@@ -0,0 +1,189 @@
+"""Matplotlib ecdfplot."""
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.stats import uniform, binom
+
+from arviz.plots.plot_utils import _scale_fig_size
+from arviz.plots.backends.matplotlib import backend_kwarg_defaults, create_axes_grid, backend_show
+
+
+def plot_ecdf(
+    values,
+    values2,
+    distribution,
+    difference,
+    pit,
+    confidence_bands,
+    granularity,
+    num_trials,
+    alpha,
+    figsize,
+    ecdf_fill,
+    ax,
+    show,
+    backend_kwargs,
+):
+    """Matplotlib ecdfplot."""
+    if backend_kwargs is None:
+        backend_kwargs = {}
+
+    backend_kwargs = {
+        **backend_kwarg_defaults(),
+        **backend_kwargs,
+    }
+
+    (figsize, _, _, _, _, _) = _scale_fig_size(figsize, None)
+    backend_kwargs.setdefault("figsize", figsize)
+    backend_kwargs["squeeze"] = True
+
+    if ax is None:
+        _, ax = create_axes_grid(1, backend_kwargs=backend_kwargs)
+
+    n = len(values)
+
+    if confidence_bands:
+        if pit:
+            x = np.linspace(1 / granularity, 1, granularity)
+            z = x
+            values = (
+                distribution(values)
+                if distribution
+                else compute_ecdf(values2, values) / len(values2)
+            )
+        else:
+            x = np.linspace(values[0], values[-1], granularity)
```

Comment:
@sethaxen even if we change this line and keep x as `values` still our main concern is the `get_gamma` function due to which the plot is slow apart from that the plot does not take even 0.2s. But that function would not be affected even if we set x as `values` rather than equally spaced points between values[0] and values[-1]. In get_gamma we need to get the ecdf values for the uniform sample w.r.t `z`. Please take a look into this and tell me if I am correct or not. So I think what @OriolAbril was saying would speed up some parts of the code apart from get_gamma but they are already fast so we wont achieve anything.

Reply:
> the plot does not take even 0.2s

But this is only one plot, imagine a user wants to plot the ecdf of 20 variables, also not sure how many samples you used for benchmarking, but we could easily have >1e6 samples together with 20, 40 or more variables to plot. In those cases the waiting time will be more than noticeable and it probably doesn't need to be. We want ArviZ to provide tools for direct exploratory analysis but also to provide building blocks for custom plots and workflows, which could even be online diagnostics during mcmc sampling or interactive tools for model checking or prior elicitation. 

> But that function would not be affected even if we set x as values rather than equally spaced points between values[0] and values[-1]. In get_gamma we need to get the ecdf values for the uniform sample w.r.t z

I don't have the algorithm in my mind, so I don't really know if `get_gamma` needs to evaluate the uniform samples on the same `z` as the ecdf of the provided samples, in which case we'd have 1 ecdf computed with the fast sorting algorithm for num_trials computed with the while algorithm, and the difference in computation time would be completely negligible, but I think this misses the point and the goal I was commenting above. In many cases we'll want an ecdf without bands (we already do that in plot_ppc for example) and again we can have many ecdfs to plot with potentially many samples.

Reply:
So @OriolAbril  I did an analysis of the plot_ecdf with and without computing the bands with a sample size of 1000, granularity(number of evaluation points): 100, num_trials:500. 
Now without confidence bands(no compute_gamma) the entire plot takes around `0.0117s` for plotting ecdf of the sample or the PIT. 
Now when `confidence_bands=True` then the entire plot takes `7.7878s` and the compute_gamma in itself takes `7.7758s`. Now we were talking about using evaluation points s.t we could find compute ecdfs as suggested in `plot_ppc`, but after analysing I found that the `compute_ecdf` which is used inside the `compute_gamma` takes just `0.0812s` which is negligible as compared to the time taken by entire compute_gamma, so rather than thinking to optimize the `compute_ecdf` we should think of optimizing the part of `compute_gamma` apart from `compute_ecdf`.

Reply:
Also I will use `np.empty` rather than initializing the list as empty and then filling it during the loop. Thanks for this